### ISSUE TITLE
[BUG] Specific Metallic Skins don't look shiny in first person if a strange stat clock is attached.

### ISSUE THREAD

**Issue by Wosheereal** [mastercomfig/tf2-patches#407]
### Description
https://www.reddit.com/r/tf2/comments/h0eojx/bug_with_the_metal_war_paints_and_a_way_to_fix_it/
this reddit post has a good idea of what the bug is.

### Checklist

- [X] None of the open *or* closed issues document this problem.
- [X] This is a bug, not a new idea.
- [X] This is objectively a bug, and not something that people commonly disagree on.
- [ ] This issue occurs on the latest Team Comtress release, or the `master` branch.
- [X] This issue occurs on the latest vanilla TF2 update.

- [ ] This bug is in the code of the game, not in the textures, sounds, or other resources.
- [X] This issue is listed on the [Source 1 official bugtracker](https://github.com/ValveSoftware/Source-1-Games/issues).
- [ ] This issue occurs with DirectX level 100.
- [X] This issue occurs when using no custom content.
- [ ] This issue occurs when using `start_tf2.bat` (or `start_server.bat`) with no custom launch options.
- [ ] This issue occurs when using a listen server (using the `map` command ingame).
- [ ] This issue occurs when using a dedicated server.
- [ ] This bug is in the internal procedures or tooling (e.g. the build process).

### Reproduction

1. Own a skin that uses albedo texturing. (Coffin Nail, Macaw Masked, Yeti Coated, Dressed to Kill.)
2.
3.

#### Client Settings

- *Server or client:* 
- *Operating system:* 
- *Team Comtress version:* 
- *Custom content:* 
- *Launch options:* 
#### Driver Version

- *Graphics driver:* 
- *DirectX level:* <95>
#### Hardware Specs

- *CPU model:* <Intel I5 @ 3.5 ghz>
- *GPU model:* <MSI GTX 970 4GB>
- *RAM amount:* <32 GB>
- *TF2 install location:* <C:/tf2>
- *TC2 install location:* <N/A>
- *TC2 installation method:* <Basic>

**Comment by treacherousfiend** [comment]
seems simple enough to fix, i'll see what I can do

**Comment by treacherousfiend** [comment]
So a bit of an update, It seems like the stat-clocks do a material check/change every time the number on it changes.
Whereas the weapon skin does a material override, but seems like it can only do that once or twice when its loaded?
I unfortunately do not have a weapon skin that uses the metal material and has a statclock on it, so if you could send screenshots of said weapon right as you join a server, and then a screenshot after you've gotten a kill? (It doesn't need to be right after you get the kill).

Also I have a question, the reddit post also mentions this happening to painted weapons with killstreak sheens (specialized killstreak weapons). Once again I cannot test this, but I also cannot find anything material related regarding killstreaks, as its an overlay that is put on top of the weapon, and does not use a material.
Basically, does this bug actually happen with specialized killstreak weapons that use the metal material.

**Comment by Wosheereal** [comment]
that'll depend, how big is the team comtress 2 install?

**Comment by treacherousfiend** [comment]
No just test it in live, I haven't fixed it yet

**Comment by Wosheereal** [comment]
I dont think it affects specialized weapons, i think it is mostly a stat clock issue.

**Comment by treacherousfiend** [comment]
alright thanks for the confirmation, i'll focus on specifically statclocks.

**Comment by Wosheereal** [comment]
![stuff](https://user-images.githubusercontent.com/29549872/95138946-48ffbe80-0720-11eb-95d2-a67caf2b226d.jpg)
![e](https://user-images.githubusercontent.com/29549872/95138951-4b621880-0720-11eb-80c0-33dd737295df.jpg)

**Comment by Wosheereal** [comment]
notice how the first person model does not have the shiny effect the third person model does?

**Comment by treacherousfiend** [comment]
interesting that its only in first person, but that should actually help narrow it down quite a bit

**Comment by treacherousfiend** [comment]
Probably should've mentioned this earlier, but I don't think I can fix this as of right now. Currently war paints do not work _at all_ in TC2, so there's not really much we can do about this or other war paint bugs rn.

**Comment by 1556197** [comment]
@treacherousfiend 
Hello, I'm the author of the original reddit post.

1. I'd like to note that this bug occurs on original gun mettle metallic weapon collections (dressed to kill, coffin nail) which have been in the game since 2015, before the war paints were added, so it must be in the 2017 tf2 version.

2. Indeed, this bug apperars only in first person view

3. This bug shows since the first spawn and doesn't depend on kills

4. Specialized killstreak sheens influencing this bug is unconfirmed, I've seen some complaints, but it may not be true. The main  trigger is stat clock.

**Comment by treacherousfiend** [comment]
Thanks for clearing some things up, unfortunately when I say that war paints don't work, I mean all decorated weapons don't work.
It seems like the code leak that we are using has a good bit of the war paint code there, but it has broken all decorated weapons.

However maybe some good news? I was able to force itemtest to give me a weapon that applies this skin and it does seem to apply the metal material despite not loading the actual skin? (This seems to be an issue with itemtest in general, not just with TC2)

It's starting to look like maybe I can fix this despite the skins not working. I'll look into forcing a statclock onto it as well.

**Comment by 1556197** [comment]
@treacherousfiend 
Well, that's good there is some chance. Thank you for trying.

**Comment by 1556197** [comment]
@treacherousfiend 
Did you manage to load the statclock on itemtest?

**Comment by 1556197** [comment]
@treacherousfiend 
I've got new information that stat clocks also ruin ubercharge covering.

**Comment by treacherousfiend** [comment]
I believe thats a bug with weapon skins in general. unfortunately, I do not have the time or resources to work on this right now, so this will be on the backburner for a while.

**Comment by 1556197** [comment]
@treacherousfiend ,
I'm currently looking into this problem in the source code. I want to compare the viewmodel code to worldmodel. So far I found .cpp file that renders dropped weapons and .cpp file that renders loadout weapons in the backpack. Unfortunately, code commentaries are very scarce.
Could you give me a hint? In case you know, what .cpp file is responsible for rendering weapons in a viewmodel and what .cpp file is responsible for rendering weapons in 3-rd person worldmodel?

**Comment by 1556197** [comment]
This got fixed in tf2.
Complete bug description and solution is here:
https://github.com/ValveSoftware/Source-1-Games/issues/3278
